### PR TITLE
Guides Navigation Redesign

### PR DESCRIPTION
**Pull request by bryanfriedman**
This is a pretty big change to the navigation and metadata around Guides but allows for us to create "subsections" within a Guides section which gives us the ability to more easily include ordered/multi-part guides, and also adds more flexibility with  short/long versions of the titles. Specifically, this change encompasses:

- Removes the confusing and not well-used `sortTitle` field and the weird backward construct of "Foo: What Is It?" or "Bar: Getting Started"
- Introduces two new fields for Guides pages:
   - `subsection` defines a Guide as a subsection that can contain one or more guides under it
   - `parent` defines the parent subsection for guides that should live under it
- Uses the Hugo-supported `linkTitle` field to specify a "short" title to appear in the Navigation menu, where the `title` will still appear on the page and for SEO purposes
- Changes the sidebar tree view to:
   - Utilize this new metadata to display a two-level hierarchy
   - Hide other Guides sections other than the current one
   - Include icon and "All Guides" link
   - Change selected/active styles to a cleaner look (just the green, no background color)
- Adds index of boxes on the top-level page for each Guides section
- Adds "On This Page" to top of ToC on the right side of guides pages to clarify what the inline links are

Importantly, none of this requires changes to the directory structure or URL paths of existing content. The guides sections still match the topics. 

@jacksoncvm I worked on this with Rachel and sent for her to look at as well, but please take a look in case I did something wrong with the CSS or HTML

@tybritten @BrianMMcClain @tiffanyfay Thoughts, feedback?

(Note: Some work is still required to fill in content gaps such as top-level subsection content as in Remote Tips, Tekton, for example. Still need to document new frontmatter fields in README and updated archetypes as well assuming we go this route.)

### CODE REVIEW

**Review by tiffanyfay — APPROVED** [review]
I like this. The only thing I had was basically what you said "Some work is still required to fill in content gaps such as top-level subsection content". Since some of them are clicked on, but the main page isn't necessarily a high level or the right content to fit as a general/first page.

**Comment by bryanfriedman** [comment]
> I like this. The only thing I had was basically what you said "Some work is still required to fill in content gaps such as top-level subsection content". Since some of them are clicked on, but the main page isn't necessarily a high level or the right content to fit as a general/first page.

Thanks @tiffanyfay. @BrianMMcClain is working on content for the Tekton main page. I think everything else is covered but did you notice anything else that was weird or needs a different main page?

File: themes/docsy/layouts/partials/toc.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 {{ if not .Params.notoc }}
 {{ with .TableOfContents }}
 {{ if ge (len .) 200 }}
-<h5 class="mb-4">On This Page</h5>
+<h5 class="mt-6 mb-4">On This Page</h5>
```
Comment (tiffanyfay) [inline themes/docsy/layouts/partials/toc.html:5]: Can it also be aligned with the stuff under it? It's further to the left. Thanks.

Comment (bryanfriedman) [reply themes/docsy/layouts/partials/toc.html:5]: Yeah I think the quote box comes with an indent so maybe that's what's showing up?

**Review by tybritten — APPROVED** [review]

**Pull request merged**